packages/core/src/core/deploy.ts

```diff
@@ -52,6 +55,26 @@ export type StepEvents = {
   failed: (errorText?: string) => void
 }
 
+const updatePlanElement = (item: PlanItem, appliedChanges: ReadonlyArray<Change>): void => {
+  const elemIDtoChangeElement = _.keyBy(
+    [...item.items.values()].map(getChangeElement),
+    changeElement => changeElement.elemID.getFullName()
+  )
+  appliedChanges.forEach(change => {
+    const updatedElement = getChangeElement(change)
+    if (!isRemovalDiff(change)) {
+      const itemChangeData = elemIDtoChangeElement[change.data.after.elemID.getFullName()]
+      if (itemChangeData !== undefined) {
+        const detailedChanges = detailedCompare(itemChangeData, updatedElement)
+        detailedChanges.forEach(detailedChange => {
+          const data = isRemovalDiff(detailedChange) ? undefined : detailedChange.data.after
+          setPath(itemChangeData, detailedChange.id, data)
+        })
+      }
+    }
+  })
```

Comment:
style nit - this deep nesting of code makes it hard to read IMO
```suggestion
  appliedChanges
    .filter(isAdditionOrModificationDiff)
    .map(getChangeElement)
    .forEach(updatedElement => {
      const planElement = planElementsByID[updatedElement.elemID.getFullName()]
      if (planElement !== undefined) {
        applyDetailedChanges(planElement, detailedCompare(planElement, updatedElement))
      }
    }
  })
```